core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -537,6 +537,7 @@ class KafkaController(val config: KafkaConfig,
 
   private def removeMetrics(): Unit = {
     KafkaController.MetricNames.foreach(metricsGroup.removeMetric)
+    controllerChannelManager.removeMetrics()
```

Comment:
KafkaController#shutdown() owns the responsibility of closing controller channel manager. ControllerChannelManager should own the responsibility of closing it's own metrics as part of it's shutdown()

Reply:
Since the current ControllerChannelManager only removes the tagged metrics (that is, gaugeMetricNameWithTag and timerMetricNameWithTag) of all brokers in brokerStateInfo during shutdown, but omits TotalQueueSizeMetricName, I added the removeMetrics method to remove it during broker shutdown. Do you mean to combine the two into the shutdown method of ControllerChannelManager? @divijvaidya